src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -187,6 +197,9 @@ public function loadClass($class)
                     if ($parent && isset(self::$deprecated[$parent]) && strncmp($ns, $parent, $len)) {
                         @trigger_error(sprintf('The %s class extends %s that is deprecated %s', $name, $parent, self::$deprecated[$parent]), E_USER_DEPRECATED);
                     }
+                    if ($parent && isset(self::$final[$parent]) && strncmp($ns, $parent, $len)) {
+                        @trigger_error(sprintf('Extending the class %s in %s is deprecated and may break in the next major release.', $parent, $name), E_USER_DEPRECATED);
```

Comment:
That would imply changing the annotation to something like `@final extending is deprecated since 42.3, to be made final in 43` and parse that… too complicated IMO and the value of this piece of information is questionable.